zipkin-web/src/main/resources/app/js/page/default.js

```diff
@@ -42,8 +44,22 @@ define(
         window.document.title = 'Zipkin - Index';
         DefaultData.attachTo(document);
 
+        const query = queryString.parse(window.location.search);
+
         this.on(document, 'defaultPageModelView', function(ev, modelView) {
-          this.$node.html(defaultTemplate(modelView));
+          const limit = query.limit || window.config.queryLimit;
+          const minDuration = query.minDuration;
+          const endTs = query.endTs || new Date().getTime();
+          const serviceName = query.serviceName || '';
+          const annotationQuery = query.annotationQuery || '';
+          this.$node.html(defaultTemplate({
+            limit,
+            minDuration,
+            endTs,
+            serviceName,
+            annotationQuery,
+            ...modelView
```

Comment:
object spread syntax: for each key in modelView, put the key and its value into this object.